### PR TITLE
PHOENIX-6317 Add OWASP dependency check for connectors

### PR DESCRIPTION
**Pull request by richardantal**

### CODE REVIEW

**Comment by stoty** [comment]
:broken_heart: **-1 overall**






| Vote | Subsystem | Runtime | Comment |
|:----:|----------:|--------:|:--------|
| +0 :ok: |  reexec  |   5m 20s |  Docker mode activated.  |
||| _ Prechecks _ |
| +1 :green_heart: |  dupname  |   0m  0s |  No case conflicting files found.  |
| +1 :green_heart: |  @author  |   0m  0s |  The patch does not contain any @author tags.  |
| -1 :x: |  test4tests  |   0m  0s |  The patch doesn't appear to include any new or modified tests. Please justify why no new tests are needed for this patch. Also please list what manual steps were performed to verify this patch.  |
||| _ master Compile Tests _ |
| +0 :ok: |  mvndep  |  10m 54s |  Maven dependency ordering for branch  |
| +1 :green_heart: |  mvninstall  |  20m 12s |  master passed  |
| +1 :green_heart: |  compile  |   2m 53s |  master passed  |
| +1 :green_heart: |  javadoc  |   1m 28s |  master passed  |
||| _ Patch Compile Tests _ |
| +0 :ok: |  mvndep  |   0m 23s |  Maven dependency ordering for patch  |
| +1 :green_heart: |  mvninstall  |  20m 47s |  the patch passed  |
| +1 :green_heart: |  compile  |   3m  2s |  the patch passed  |
| +1 :green_heart: |  javac  |   3m  2s |  the patch passed  |
| +1 :green_heart: |  whitespace  |   0m  0s |  The patch has no whitespace issues.  |
| +1 :green_heart: |  xml  |   0m  3s |  The patch has no ill-formed XML file.  |
| +1 :green_heart: |  javadoc  |   1m 28s |  the patch passed  |
||| _ Other Tests _ |
| -1 :x: |  unit  | 111m 21s |  root in the patch failed.  |
| -1 :x: |  asflicense  |   0m 25s |  The patch generated 7 ASF License warnings.  |
|  |   | 178m 42s |   |


| Reason | Tests |
|-------:|:------|
| Failed junit tests | phoenix.hive.HiveTezIT |
|   | phoenix.hive.HiveMapReduceIT |


| Subsystem | Report/Notes |
|----------:|:-------------|
| Docker | ClientAPI=1.41 ServerAPI=1.41 base: https://ci-hadoop.apache.org/job/Phoenix/job/Phoenix-Connectors-PreCommit-GitHub-PR/job/PR-57/1/artifact/yetus-general-check/output/Dockerfile |
| GITHUB PR | https://github.com/apache/phoenix-connectors/pull/57 |
| Optional Tests | dupname asflicense javac javadoc unit xml compile |
| uname | Linux 43049ff4feb3 4.15.0-142-generic #146-Ubuntu SMP Tue Apr 13 01:11:19 UTC 2021 x86_64 x86_64 x86_64 GNU/Linux |
| Build tool | maven |
| Personality | dev/phoenix-connectors-personality.sh |
| git revision | master / a732b66 |
| Default Java | Private Build-1.8.0_242-8u242-b08-0ubuntu3~16.04-b08 |
| unit | https://ci-hadoop.apache.org/job/Phoenix/job/Phoenix-Connectors-PreCommit-GitHub-PR/job/PR-57/1/artifact/yetus-general-check/output/patch-unit-root.txt |
|  Test Results | https://ci-hadoop.apache.org/job/Phoenix/job/Phoenix-Connectors-PreCommit-GitHub-PR/job/PR-57/1/testReport/ |
| asflicense | https://ci-hadoop.apache.org/job/Phoenix/job/Phoenix-Connectors-PreCommit-GitHub-PR/job/PR-57/1/artifact/yetus-general-check/output/patch-asflicense-problems.txt |
| Max. process+thread count | 1793 (vs. ulimit of 30000) |
| modules | C: phoenix-hive-base . U: . |
| Console output | https://ci-hadoop.apache.org/job/Phoenix/job/Phoenix-Connectors-PreCommit-GitHub-PR/job/PR-57/1/console |
| versions | git=2.7.4 maven=3.3.9 |
| Powered by | Apache Yetus 0.12.0 https://yetus.apache.org |


This message was automatically generated.

**Comment by stoty** [comment]
Does the check throw up anything interesting that we should update ?

File: phoenix-hive-base/pom.xml [hunk 1]
```diff
@@ -589,6 +589,7 @@
       <dependency>      
         <groupId>org.apache.calcite</groupId>
         <artifactId>calcite-core</artifactId>
+        <version>1.10.0</version>
```
Comment (stoty) [inline phoenix-hive-base/pom.xml:592]: Why was this necessary for this patch ?

Comment (richardantal) [reply phoenix-hive-base/pom.xml:592]: I ran  `mvn clean compile test-compile site -Dspotbugs.site` without this version and I got the following exception:
`Execution default-site of goal org.apache.maven.plugins:maven-site-plugin:3.7.1:site failed: For artifact {org.apache.calcite:calcite-core:null:jar}: The version cannot be empty.`
 So I ran `mvn dependency:tree` to check the version in phoenix4-hive and added it.

Comment (stoty) [reply phoenix-hive-base/pom.xml:592]: This depends on the Hive version.
Could you extract this into a property to make it easier to override ?

**Review by stoty — APPROVED** [review]
+1 LGTM
